core/src/main/java/hudson/model/Queue.java

```diff
@@ -430,31 +452,39 @@ public synchronized void load() {
             }
         } catch (IOException e) {
             LOGGER.log(Level.WARNING, "Failed to load the queue file " + getXMLQueueFile(), e);
+        } finally {
+            updateSnapshot();
```

Comment:
Should this not be at the end of the main part of the `try` block, rather than in the `finally` block? What if it throws a `RuntimeException`?